# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ImageTypes.h
# Commit: 35d86243cb45
# Full Hash: 35d86243cb45f7e02dd6ec0eea04cd62c2a70196
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2024-09-16 09:14:26
# Regressor Bug: 1908245
# File Overlap Count: 1
# Description:
#   Bug 1908245 - Add support for missing requestVideoFrameCallback parameters. r=media-playback-reviewers,webidl,padenot,saschanaz
#   
#   This patch adds support for exposing the capture time, receive time,
#   processing duration and RTP timestamp parameters on the callback for
#   rVFC for WebRTC related elements.
# ==============================================================================

diff -r 7d143d92bf2a -r 35d86243cb45 gfx/layers/ImageTypes.h
--- a/gfx/layers/ImageTypes.h	Sun Sep 15 20:03:40 2024 +0000
+++ b/gfx/layers/ImageTypes.h	Sun Sep 15 21:49:38 2024 +0000
@@ -8,6 +8,9 @@
 #define GFX_IMAGETYPES_H
 
 #include <stdint.h>  // for uint32_t
+#include "mozilla/Maybe.h"
+#include "mozilla/TimeStamp.h"
+#include "mozilla/Variant.h"
 
 namespace mozilla {
 
@@ -112,12 +115,20 @@
 
 namespace layers {
 
-typedef uint32_t ContainerFrameID;
+using ContainerFrameID = uint32_t;
 constexpr ContainerFrameID kContainerFrameID_Invalid = 0;
 
-typedef uint32_t ContainerProducerID;
+using ContainerProducerID = uint32_t;
 constexpr ContainerProducerID kContainerProducerID_Invalid = 0;
 
+// int64_t represents a WebRTC NTP timestamp.
+using ContainerCaptureTime = Variant<Nothing, TimeStamp, int64_t>;
+
+// int64_t represents a WebRTC Realtime timestamp.
+using ContainerReceiveTime = Maybe<int64_t>;
+
+using ContainerRtpTimestamp = Maybe<uint32_t>;
+
 }  // namespace layers
 
 }  // namespace mozilla